arbnode/inbox_tracker.go

```diff
@@ -598,6 +621,31 @@ func (b *multiplexerBackend) ReadDelayedInbox(seqNum uint64) (*arbostypes.L1Inco
 var delayedMessagesMismatch = errors.New("sequencer batch delayed messages missing or different")
 
 func (t *InboxTracker) AddSequencerBatches(ctx context.Context, client arbutil.L1Interface, batches []*SequencerInboxBatch) error {
+	var nextAcc common.Hash
+	var prevbatchmeta BatchMetadata
+	sequenceNumberToKeep := uint64(0)
+	if t.snapSyncConfig.Enabled {
```

Comment:
during most of a node's sync we'll be past this.. so let's have the if config something like:
```
if snapSyncConfig.Enabled && batches[0].SequenceNumber < sequenceNumberToKeep
```
That way it'll immediately be skipped during most execution.

Also - let's do it all after the if len(batches) == 0 so the code above doesn't panic :)

Reply:
done.